src/OpenTelemetry.Instrumentation.ConfluentKafka/OpenTelemetryConsumeResultExtensions.cs

```diff
@@ -0,0 +1,171 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+using System.Text;
+using OpenTelemetry;
+using OpenTelemetry.Context.Propagation;
+using OpenTelemetry.Instrumentation.ConfluentKafka;
+using OpenTelemetry.Trace;
+
+namespace Confluent.Kafka;
+
+/// <summary>
+/// <see cref="IConsumer{TKey,TValue}"/> extension methods.
+/// </summary>
+public static class OpenTelemetryConsumeResultExtensions
+{
+    /// <summary>
+    /// Attempts to extract a <see cref="PropagationContext"/> from the <see cref="ConsumeResult{TKey,TValue}"/>'s <see cref="Headers"/> property.
+    /// </summary>
+    /// <param name="consumeResult">The <see cref="ConsumeResult{TKey,TValue}"/>.</param>
+    /// <param name="propagationContext">The <see cref="PropagationContext"/>.</param>
+    /// <typeparam name="TKey">The type of key of the <see cref="ConsumeResult{TKey,TValue}"/>.</typeparam>
+    /// <typeparam name="TValue">The type of value of the <see cref="ConsumeResult{TKey,TValue}"/>.</typeparam>
+    /// <returns>True when a <see cref="PropagationContext"/> has been extracted from <see cref="Headers"/>, otherwise false.</returns>
+    public static bool TryExtractPropagationContext<TKey, TValue>(
+        this ConsumeResult<TKey, TValue> consumeResult,
+        out PropagationContext propagationContext)
+    {
+#if NETFRAMEWORK
+        if (consumeResult == null)
+        {
+            throw new ArgumentNullException(nameof(consumeResult));
+        }
+#else
+        ArgumentNullException.ThrowIfNull(consumeResult);
+#endif
+
+        try
+        {
+            propagationContext = ExtractPropagationContext(consumeResult.Message?.Headers);
+            return true;
+        }
+        catch
+        {
+            propagationContext = default;
+            return false;
+        }
+    }
+
+    /// <summary>
+    /// Consumes a message and creates <see href="https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md#span-kind">a process span</see> embracing the <see cref="ConsumeAndProcessMessageHandler{TKey,TValue}"/>.
+    /// </summary>
+    /// <param name="consumer">The <see cref="IConsumer{TKey,TValue}"/>.</param>
+    /// <param name="handler">A <see cref="ConsumeAndProcessMessageHandler{TKey,TValue}"/>.</param>
+    /// <param name="cancellationToken">An optional <see cref="CancellationToken"/>.</param>
+    /// <typeparam name="TKey">The type of key of the <see cref="ConsumeResult{TKey,TValue}"/>.</typeparam>
+    /// <typeparam name="TValue">The type of value of the <see cref="ConsumeResult{TKey,TValue}"/>.</typeparam>
+    /// <returns>A <see cref="ValueTask"/>.</returns>
+    public static async ValueTask<ConsumeResult<TKey, TValue>?> ConsumeAndProcessMessageAsync<TKey, TValue>(
+        this IConsumer<TKey, TValue> consumer,
+        ConsumeAndProcessMessageHandler<TKey, TValue> handler,
+        CancellationToken cancellationToken = default)
```

Comment:
As I remember it is not good practice to expose public api with defaults.
Please, cosider to use overloads with and without this token as a parameter.